### PR TITLE
Remove collection size feature

### PR DESCRIPTION
**Pull request by mjgiarlo**
While looking into #4100, we noticed that `CharacterizeJob` not only saves a file set's `original_file`, but also reindexes the file set, *and* reindexes every collection to which the file set's work belongs. It appears that this feature was added way back in the Sufia days in order to support the use case of displaying on the collection show page the total size of all files uploaded to a collection's works.

Why is this problematic?

At large scale, this means touching a potentially huge portion of the repository—iterating over all files in all works in a collection—in order to display an arguably useful string to a subset of repository users who care about such information. Doing this in `CharacterizeJob` introduces significant potential delay in the ingest pipeline, increasing how long it takes for a deposit to finish and render in the Hyrax UI (e.g., by delaying the creation of derivatives).

We propose removing this feature for now and encourage folks who need this feature to contribute it back with a better-performing, more streamlined design.

### CODE REVIEW

**Comment by mjgiarlo** [comment]
> Do we need to add logic to deprecate the `#size` method usage? (e.g. have a `#size` method for collections that returns `n/a`)

I defer to @no-reply and others who have a hyrax in this hunt.

**Comment by no-reply** [comment]
:100: to this! thanks.

i do think it would be nice to make the public methods emit deprecation warnings and return hard-coded & compatible null-ish values

File: app/models/concerns/hyrax/collection_behavior.rb [hunk 1]
```diff
@@ -111,18 +111,15 @@ def collection_type_gid_document_field_name
...
+    #   hard-coded null implementation
+    # @return [Fixnum] 0
     def bytes
-      return 0 if member_object_ids.empty?
-
-      raise "Collection must be saved to query for bytes" if new_record?
-
-      # One query per member_id because Solr is not a relational database
-      member_object_ids.collect { |work_id| size_for_work(work_id) }.sum
+      Deprecation.warn("#bytes has been deprecated for removal in Hyrax 4.0.0; " \
```
Comment (mjgiarlo) [inline app/models/concerns/hyrax/collection_behavior.rb]: I'm a little surprised Rubocop didn't flag this for using double-quotes. I don't care deeply about this, but thought I'd raise it to see if you do.

File: app/models/concerns/hyrax/collection_behavior.rb [hunk 2]
```diff
@@ -111,18 +111,15 @@ def collection_type_gid_document_field_name
...
+                       "The implementation of the indexed Collection size " \
+                       "feature extremely inefficient, so it has been removed. " \
```
Comment (mjgiarlo) [inline app/models/concerns/hyrax/collection_behavior.rb]: Typo. I think we need an `is ` between `feature` and `extremely inefficient`

File: app/presenters/hyrax/collection_presenter.rb [hunk 3]
```diff
@@ -41,17 +41,20 @@ def collection_type
...
       self.class.terms.select { |t| self[t].present? }
     end
 
+    ##
+    # @param [Symbol] key
+    # @return [Object]
     def [](key)
       case key
-      when :size
+      when size
```
Comment (mjgiarlo) [inline app/presenters/hyrax/collection_presenter.rb]: I believe this ought to be a symbol, `:size`.

File: app/presenters/hyrax/collection_presenter.rb [hunk 4]
```diff
@@ -60,8 +63,16 @@ def [](key)
...
     end
 
+    # @deprecated to be removed in 4.0.0; this feature was replaced with a
+    #   hard-coded null implementation
+    # @return [String] 'unknown'
     def size
-      number_to_human_size(@solr_document['bytes_lts'])
+      Deprecation.warn("#size has been deprecated for removal in Hyrax 4.0.0; " \
+                       "The implementation of the indexed Collection size " \
+                       "feature extremely inefficient, so it has been removed. " \
```
Comment (mjgiarlo) [inline app/presenters/hyrax/collection_presenter.rb]: Same comments as above re: double- vs. single-quotes and the need to add `is `.

File: spec/presenters/hyrax/collection_presenter_spec.rb [hunk 5]
```diff
@@ -126,12 +122,6 @@
     it { is_expected.to eq ['adc12v'] }
   end
 
-  describe "#size", :clean_repo do
```
Comment (mjgiarlo) [inline spec/presenters/hyrax/collection_presenter_spec.rb]: Should we tweak these tests to verify that the hard-coded value is returned? And/or that `Deprecation` receives `:warn`?

File: app/presenters/hyrax/collection_presenter.rb (hunk 3 above)
Comment (jeremyf) [reply app/presenters/hyrax/collection_presenter.rb]: Agreed, this should be `:size`

File: app/models/concerns/hyrax/collection_behavior.rb (hunk 1 above)
Comment (jcoyne) [reply app/models/concerns/hyrax/collection_behavior.rb]: Bixby don't care about how many quotes you use.

**Comment by elrayle** [comment]
I know this is already merged, but I just want to add my kudos to @mjgiarlo (and any others working on this PR).  Always happy to see performance enhancements.

**Pull request merged**